libpromises/conversion.c

```diff
@@ -409,7 +409,7 @@ long IntFromString(const char *s)
 
     remainder[0] = '\0';
 
-    sscanf(s, "%lld%c%s", &a, &c, remainder);
+    sscanf(s, "%lld%c%4095s", &a, &c, remainder);
```

Comment:
Sure, converted remainder to char.

Not sure what you mean by checking sscanf() return value being the problem, as arbitrary length written to the string reminder is now avoided?